pkg/osquery/tables/gsettings/gsettings.go

```diff
@@ -0,0 +1,282 @@
+package gsettings
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	osquery "github.com/kolide/osquery-go"
+	"github.com/kolide/osquery-go/plugin/table"
+	"github.com/pkg/errors"
+)
+
+type Table struct {
+	client *osquery.ExtensionManagerClient
+	logger log.Logger
+}
+
+func TablePlugin(client *osquery.ExtensionManagerClient, logger log.Logger) *table.Plugin {
+	columns := []table.ColumnDefinition{
+		// TODO: maybe need to add 'path' for relocatable schemas..
+		table.TextColumn("domain"),
+		table.TextColumn("key"),
+		table.TextColumn("type"),
+		table.TextColumn("value"),
+		table.TextColumn("description"),
+	}
+
+	t := &Table{
+		client: client,
+		logger: logger,
+	}
+
+	return table.NewPlugin("kolide_gsettings", columns, t.generate)
+}
+
+func (t *Table) generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	var results []map[string]string
+
+	domainQ, ok := queryContext.Constraints["domain"]
+	if !ok || len(domainQ.Constraints) == 0 {
+		return nil, errors.New("the kolide_gsettings table requires a domain to be specified")
+	}
+
+	for _, domainConstraint := range domainQ.Constraints {
```

Comment:
Want to use `tablehelpers.GetConstraints` here, or stick with this?

Reply:
oh yea, I'll use the tablehelpers